Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -141,9 +141,29 @@ IPCCommandResult NetKDRequest::IOCtl(const IOCtlRequest& request)
     break;
 
   case IOCTL_NWC24_REQUEST_SHUTDOWN:
-    // if ya set the IOS version to a very high value this happens ...
-    INFO_LOG(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_REQUEST_SHUTDOWN - NI");
+  {
+    if (request.buffer_in == 0 || (request.buffer_in & 3) != 0 || request.buffer_in_size < 8 ||
```

Comment:
While it's fairly clear what this is checking for, I'd suggest using `x % 4 != 0` rather than `(x & 3) != 0` to make it even more obvious (bonus: it removes the need for the parentheses)